docs/06-concepts/11-authentication/05-custom-overrides.md

```diff
@@ -177,3 +177,18 @@ var token = await client.authenticationKeyManager?.get();
 ```
 
 The above example retrieves the token from the key manager and stores it in the `token` variable.
+
+### Advanced authentication schemes
+
+By default Serverpod will pass the authentication token from client to server in accordance with the HTTP "authorization" header standard with the "basic" scheme name and encoding. This is securely transferred as the connection is TLS encrypted.
+
+The default implementation encodes and wraps the user-provided token in a "basic" scheme which is automatically unwrapped on the server side before being handed to the user-provided authentication handler described above. *In other words the default transport implementation is "invisible" to user code.*
+
+If you are implementing another authentication scheme, for example OAuth 2 using bearer tokens, you should override the default method `toHeaderValue` of `AuthenticationKeyManager`. This client-side method converts the authentication key to the format that shall be sent as a transport header to the server.
+
+You will also need to implement the `AuthenticationHandler` accordingly, in order to process that header value server-side.
```

Comment:
Question: Could it be a good idea to provide an actual short example of the case you mentioned?